src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -677,7 +677,7 @@ def load_arguments(self, _):
         with self.argument_context(scope, arg_group='Network') as c:
             c.argument('vnet_name', help='Name of the virtual network when creating a new one or referencing an existing one.')
             c.argument('vnet_address_prefix', help='The IP address prefix to use when creating a new VNet in CIDR format.')
-            c.argument('subnet', help='The name of the subnet when creating a new VNet or referencing an existing one. Can also reference an existing subnet by ID. If omitted, an appropriate VNet and subnet will be selected automatically, or a new one will be created.')
+            c.argument('subnet', help='The name of the subnet when creating a new VNet or referencing an existing one. Can also reference an existing subnet by ID. If omitted, an appropriate VNet and subnet will be selected automatically only if VNet is also omitted, or a new one will be created if not exists.')
```

Comment:
This doesn't seem very clear to me: If omitted, an appropriate VNet and subnet will be selected automatically only if VNet is also omitted, or a new one will be created if not exists.

My understanding is if subnet is ommited, we will use a default name. Then the process is the same as subnet not ommited?

Reply:
The logic of selecting automatically only happens when both vnet and subnet are omitted. And the auto create subnet logic happens when the subnet not exists, regardless where the value come from (user specified or generate from default format).

Reply:
Then `A new one will be created if not exists.` should be in a separate sentence regardless of whether subnet is omitted.

Reply:
Actually a `new one will be created` is already mentioned in the first sentence. So don't bother to change.